cgi-bin/dashboard-aii.cgi

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/perl -T
+# ${license-info}
+# ${developer-info
+# ${author-info}
+# ${build-info}
+
+use Config::Simple;
+use JSON::XS;
+use Switch;
+use CGI;
+
+# Configuration file to boot from HD. Following pxelinux convention
+# it must be called default so if the <IpAddressInHex> link
+# is missing the node will boot from the hard disk
+my $boothd = "default";
+
+# Quattor profiles
+my $web_root = "/var/www/html";
+my $profiles_path = "profiles";
+my $profiles_dir = $web_root.'/'.$profiles_path;
+
+# Read AII configuration
+$cfg = new Config::Simple();
+$cfg->read('/etc/aii/aii-shellfe.conf');
+
+# PXE Linux directory
+my $pxelinux_dir = $cfg->param('nbpdir');
+
+# Profile prefix
+my $profile_prefix = $cfg->param('profile_prefix') eq 0 ? '' : $cfg->param('profile_prefix');
+
+# cdb url
+my $cdb_url = $cfg->param('cdburl');
+
+# Number of different boot types
+my %boot_type;
+
+# Array of available pxelinux configurations
+my @cfg;
+
+# Data collected from the form
+my %nodes_cfg;
+
+# Profiles .json available
+my (@profiles);
+
+################################################################
+sub GetHexAddr {
+################################################################
+# Get the hostname or an IP address and return the IP address in
+# hex (as required by pxelinux)
+
+  # The 4th field is an array of the IP address of this node
+  my @all_address;
+  @all_address =(gethostbyname($_[0]))[4];
+  if ($#all_address < 0) { # The array is empty
+    return ;
+  }
+  # We unpack the IP address
+  my @tmp_address = unpack('C4',$all_address[0]);
+  my @result;
+  $result[0]=sprintf ("%02X%02X%02X%02X",$tmp_address[0], $tmp_address[1],
+                                         $tmp_address[2], $tmp_address[3]);
+  $result[1]=sprintf ("%u.%u.%u.%u",$tmp_address[0], $tmp_address[1],
+                                    $tmp_address[2], $tmp_address[3]);
+  return @result;
+}
+
+
+#########################################################################
+sub Initialize {
+#########################################################################
+
+  @profiles=();
+
+  opendir(DIR,$profiles_dir);
+
+  # find all profiles in directory
+  push @profiles,map { s/\.json$//; s/^$profile_prefix//; $_ .=''; } sort(grep(/\.json$/, readdir(DIR)));
```

Comment:
OK for json.gz.
The map is to extract the hostname from the profile filename. (remove .json and $profile_prefix)